### PR TITLE
[SONARMSBR-98] Added upgrade information

### PR DESCRIPTION
**Pull request by duncanp-sonar**
I've added the upgrade information as an appendix.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -217,14 +218,17 @@ While preparing a Virtual Machine that will host SonarQube database, portal and/
 		sonar.jdbc.url=jdbc:jtds:sqlserver://localhost;databaseName=sonar;integratedSecurity=true;”
 		```
 
-6. **Download and install latest SonarQube C# plugin**
+6. <a name="installLatestPluginAnchor"></a>**Download and install latest SonarQube C# plugin**
+
 	- Download the latest sonar-csharp-plugin-X.Y.jar. At the time of writing, all versions of the C\# plugin are available from the [C\# Plugin](http://redirect.sonarsource.com/plugins/csharp.html) page, on the SonarQube site.
-	- MSBuild and TeamBuild integration is supported since version 4.0 of the plugin.
+	- Use version 4.1 or higher of the plugin (version 4.0 of the plugin supported the pre-release version of SonarQube MSBuild Runner).
```
Comment (duncanp-sonar) [inline README.md]: Miscellaneous change: we don't want users to download v4.0.

File: README.md [hunk 2]
```diff
@@ -217,14 +218,17 @@ While preparing a Virtual Machine that will host SonarQube database, portal and/
 		sonar.jdbc.url=jdbc:jtds:sqlserver://localhost;databaseName=sonar;integratedSecurity=true;”
 		```
 
-6. **Download and install latest SonarQube C# plugin**
+6. <a name="installLatestPluginAnchor"></a>**Download and install latest SonarQube C# plugin**
```
Comment (duncanp-sonar) [inline README.md:221]: This isn't a heading so we have to explicitly define an anchor that we can link to.

File: README.md [hunk 3]
```diff
@@ -24,6 +24,7 @@
 	- [Configure SonarQube to use Microsoft SQL Database](#configure-sonarqube-to-use-microsoft-sql-database)
 	- [Secure the SonarQube Portal](#secure-the-sonarqube-portal)
 - [Conclusion](#conclusion)
+- [Appendix 1: Upgrading from v0.9 of the SonarQube MSBuild Runner](#upgradeFrom09Anchor)
```
Comment (duncanp-sonar) [inline README.md:27]: Normally we'd just be able to create a link to a header by using the text of the header. However, the markup processor doesn't like something about this header (probably the ":" character) so I defined an explicit anchor.

File: README.md (hunk 1 above)
Comment (dbolkensteyn) [inline README.md]: @duncanpMS feel free to even drop the mention to version 4.0 altogether

File: README.md [hunk 4]
```diff
@@ -611,3 +615,45 @@ The Visual Studio ALM Rangers includes members from the Visual Studio Product gr
...
+
+Perform the following steps to upgrade from version 0.9 of the SonarQube MSBuild Runner:
+
+1. Install the new version of the C# plugin on the SonarQube server as described [above](#installLatestPluginAnchor).
+2. Install the new version of the MSBuild.SonarQube.Runner on the agent machine as described [above](#setup-of-the-msbuild-sonarqube-runner-on-the-build-agent-machine).
+3. (Optional) Migrate any additional settings from the old **sonar-runner.properties** file to the **SonarQube.Analysis.xml** file.
+	- If you had added any additional settings in the **sonar-runner.properties** file then these settings will need to be moved to the new **SonarQube.Analysis.xml** file. 
+4. Delete *SonarQube.Integration.ImportBefore.targets* from *%ProgramFiles(x86)%\MSBuild\12.0\Microsoft.Common.Targets\ImportBefore*.
+5. Upgrade any existing build definitions.
+	- The name of the executable in the **Pre-build script path** and the **Post-test script path** fields should be changed from *SonarQube.MSBuild.Runner.exe* to *MSBuild.SonarQube.Runner.exe*.
```
Comment (dbolkensteyn) [inline README.md:649]: might be worth to mention that the _begin_ and _end_ commands should be added

**Comment by dbolkensteyn** [comment]
LGTM

File: README.md (hunk 4 above)
Comment (duncanp-sonar) [inline README.md:649]: Done.

**Comment by duncanp-sonar** [comment]
Thanks @dbolkensteyn. I removed the reference to the 4.0 plugin.

**Pull request merged**